sdk/storage/azure-storage-blob/azure/storage/blob/_shared/base_client.py

```diff
@@ -237,21 +253,22 @@ def _create_pipeline(self, credential, **kwargs):
             config.transport = RequestsTransport(**kwargs)
         policies = [
             QueueMessagePolicy(),
+            config.headers_policy,
             config.proxy_policy,
             config.user_agent_policy,
             StorageContentValidation(),
-            ContentDecodePolicy(response_encoding="utf-8"),
             RedirectPolicy(**kwargs),
             StorageHosts(hosts=self._hosts, **kwargs),
             config.retry_policy,
-            config.headers_policy,
```

Comment:
just a note: headers_policy cannot be put before retry, since we need to regenerate the timestamp in header

Reply:
Thanks @xiafu-msft - I wondered about this! Why was it different between the sync and async pipelines?

Reply:
thanks Anna @annatisch if async is having different order then it's a bug...